repo/blob/gdrive/gdrive_storage.go

```diff
@@ -0,0 +1,556 @@
+// Package gdrive implements Storage based on Google Drive.
+package gdrive
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	drive "google.golang.org/api/drive/v3"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/iocopy"
+	"github.com/kopia/kopia/internal/retry"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/retrying"
+	"github.com/kopia/kopia/repo/logging"
+)
+
+const (
+	gdriveStorageType = "gdrive"
+	gdriveMimeType    = "application/x-kopia"
+
+	uploadChunkSize   = 1 << 20
+	uploadContentType = "application/octet-stream"
+	maxPrefixLen      = 26
+	queryRetryDelay   = 100 * time.Millisecond
+	queryRetryMax     = 3
+
+	// googleapi.Field values.
+	metadataFields     = "name,id,mimeType,size,modifiedTime"
+	listIDFields       = "files(name,id)"
+	listMetadataFields = "files(name,id,mimeType,size,modifiedTime)"
+)
+
+var log = logging.Module("gdrive")
+
+type gdriveStorage struct {
+	Options
+
+	client      *drive.FilesService
+	folderID    string
+	fileIDCache *FileIDCache
+}
+
+func (gdrive *gdriveStorage) GetBlob(ctx context.Context, b blob.ID, offset, length int64, output blob.OutputBuffer) error {
+	if offset < 0 {
+		return blob.ErrInvalidRange
+	}
+
+	fileID, err := gdrive.GetFileID(ctx, b)
+	if err != nil {
+		return errors.Wrapf(err, "get file id in GetBlob(%s)", b)
+	}
+
+	req := gdrive.client.Get(fileID)
+	req.Header().Set("Range", toRange(offset, length))
+	res, err := req.Context(ctx).Download()
+	
```

Comment:
There's an issue with the linter setup:
- gofumpt wants to remove it
- wsl wants to add it back

Reply:
i think you need to add empty line above 66

Reply:
That fixed it. I apparently misread the error output. Thanks!